application/temperature.py

```diff
@@ -0,0 +1,26 @@
+import httpx
+from sqlalchemy.orm import Session
+from datetime import datetime
+from application import models, schemas
+
+
+async def fetch_temperature(city_name: str):
+    response = await httpx.get(f"http://api.openweathermap.org/data/2.5/weather?q={city_name}&appid=YOUR_API_KEY")
```

Comment:
The `httpx.get` method should be replaced with `httpx.AsyncClient().get` for proper asynchronous execution. Consider using an `async with` block to manage the client context.